From f13bf7a962a854550d40af9d023f06738fe2878a Mon Sep 17 00:00:00 2001
From: Cody Xie <xkd@rock-chips.com>
Date: Thu, 20 Apr 2017 20:21:28 +0800
Subject: [PATCH] Disable root login with password.

Signed-off-by: Cody Xie <xkd@rock-chips.com>
---
 rootfs-overlay/etc/inittab | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/rootfs-overlay/etc/inittab b/rootfs-overlay/etc/inittab
index 246c94d..6a0d782 100644
--- a/rootfs-overlay/etc/inittab
+++ b/rootfs-overlay/etc/inittab
@@ -25,10 +25,10 @@ null::sysinit:/sbin/ifconfig lo up
 # now run any rc scripts
 ::sysinit:/etc/init.d/rcS
 
-#::askfirst:-/bin/sh
+::askfirst:-/bin/sh
 
 # Put a getty on the serial port
-::respawn:/sbin/getty -L ttyFIQ0 0 vt100
+#::respawn:/sbin/getty -L ttyFIQ0 0 vt100
 
 # Stuff to do before rebooting
 ::shutdown:/etc/init.d/rcK
-- 
2.7.4

