From 94281e8fa018386644336bbe59911cecc0e93ab8 Mon Sep 17 00:00:00 2001
From: lanshh <lsh@rock-chips.com>
Date: Fri, 29 Dec 2017 18:41:15 +0800
Subject: [PATCH 5/5] add user1 fatab item for

Signed-off-by: lanshh <lsh@rock-chips.com>
---
 rootfs-overlay/etc/fstab | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rootfs-overlay/etc/fstab b/rootfs-overlay/etc/fstab
index da73a40..f3fbd18 100644
--- a/rootfs-overlay/etc/fstab
+++ b/rootfs-overlay/etc/fstab
@@ -8,3 +8,4 @@ sysfs		/sys		sysfs	defaults	0	0
 debug		/sys/kernel/debug debugfs defaults	0		0
 /dev/mmcblk0p5 /param   ext4   defaults    0   0
 /dev/mmcblk0p6 /user   ext4   defaults    0   0
+/dev/mmcblk0p7 /user1   ext4   defaults    0   0
-- 
1.9.1

