From 7d6f96d1a7dac67ed75a6d4c674d3af05afc08a2 Mon Sep 17 00:00:00 2001
From: lanshh <lsh@rock-chips.com>
Date: Wed, 20 Dec 2017 16:59:39 +0800
Subject: [PATCH] anlarge rootfs partition for emmc

Signed-off-by: lanshh <lsh@rock-chips.com>
---
 setting_emmc.ini | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setting_emmc.ini b/setting_emmc.ini
index f58257e..13ce2eb 100755
--- a/setting_emmc.ini
+++ b/setting_emmc.ini
@@ -27,12 +27,12 @@ File=../rockimg/Image-cvr/kernel.img
 Name=rootfs
 Type=0x8
 PartOffset=0x5000
-PartSize=0x4800
+PartSize=0x6800
 Flag=0x0
 File=../rockimg/Image-cvr/rootfs.img
 [UserPart4]
 Name=user
 Type=0x80000000
-PartOffset=0x9800
+PartOffset=0xb800
 PartSize=0x0
 Flag=0x5
-- 
1.9.1

