From 927a206edc2e45a1dc1239ace73bb16e0f0f72c9 Mon Sep 17 00:00:00 2001
From: lanshh <lsh@rock-chips.com>
Date: Mon, 6 Nov 2017 19:37:26 +0800
Subject: [PATCH 1/2] default enable wifi and adb

Signed-off-by: lanshh <lsh@rock-chips.com>
---
 parameter.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/parameter.c b/parameter.c
index 92c2ff4..5fecf95 100644
--- a/parameter.c
+++ b/parameter.c
@@ -1240,7 +1240,7 @@ int parameter_recover(void)
     parameter.abmode = 2;
     parameter.vcam_resolution_photo = 1;
     parameter.screenoff_time = -1;
-    parameter.wifi_en = 0;
+    parameter.wifi_en = 1;
     parameter.ex = 3;
     parameter.video_mark = 1;
     parameter.gps_mark = 0;
@@ -1261,6 +1261,7 @@ int parameter_recover(void)
     parameter.video_flip = 0;
     parameter.reboot_cnt = 0;
     parameter.tv_sec = 0;
+    parameter.video_usb = 1;
 
     /*
     * add for dv
-- 
1.9.1

